libnavigation-base/src/main/java/com/mapbox/navigation/base/options/RoutingTilesOptions.kt

```diff
@@ -0,0 +1,199 @@
+     * Regenerate whenever a change is made
+     */
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as RoutingTilesOptions
+
+        if (tilesBaseUri != other.tilesBaseUri) return false
+        if (tilesDataset != other.tilesDataset) return false
+        if (tilesProfile != other.tilesProfile) return false
+        if (tilesVersion != other.tilesVersion) return false
+        if (filePath != other.filePath) return false
+        if (minDaysBetweenServerAndLocalTilesVersion
+            != other.minDaysBetweenServerAndLocalTilesVersion
+        ) return false
+
+        return true
+    }
+
+    /**
+     * Regenerate whenever a change is made
+     */
+    override fun hashCode(): Int {
+        var result = tilesBaseUri.hashCode()
+        result = 31 * result + tilesDataset.hashCode()
+        result = 31 * result + tilesProfile.hashCode()
+        result = 31 * result + tilesVersion.hashCode()
+        result = 31 * result + (filePath?.hashCode() ?: 0)
+        result = 31 * result + minDaysBetweenServerAndLocalTilesVersion.hashCode()
+        return result
+    }
+
+    /**
+     * Returns a string representation of the object.
+     */
+    override fun toString(): String {
+        return "RoutingTilesOptions(" +
+            "tilesBaseUri=$tilesBaseUri, " +
+            "tilesDataset='$tilesDataset', " +
+            "tilesProfile='$tilesProfile', " +
+            "tilesVersion='$tilesVersion', " +
+            "filePath=$filePath, " +
+            "minDaysBetweenServerAndLocalTilesVersion=$minDaysBetweenServerAndLocalTilesVersion" +
+            ")"
+    }
+
+    /**
+     * Builder for [RoutingTilesOptions]. You must choose a [filePath]
+     * for this to be built successfully.
+     */
+    class Builder {
+
+        private var tilesBaseUri: URI = URI("https://api.mapbox.com")
+        private var tilesDataset: String = "mapbox"
+        private var tilesProfile: String = "driving-traffic"
+        private var tilesVersion: String = ""
+        private var filePath: String? = null
+        private var minDaysBetweenServerAndLocalTilesVersion: Int = 56 // 8 weeks
+
+        /**
+         * Override the routing tiles base endpoint with a [URI].
+         *
+         * Expects scheme and host, for example `"https://api.mapbox.com"`.
+         */
+        fun tilesBaseUri(tilesBaseUri: URI): Builder =
+            apply {
+                val validUri = tilesBaseUri.isAbsolute &&
+                    tilesBaseUri.host.isNotEmpty() &&
+                    tilesBaseUri.fragment == null &&
+                    tilesBaseUri.path.let { it.isNullOrEmpty() || it.matches("/*".toRegex()) } &&
+                    tilesBaseUri.query == null
+                check(validUri) {
+                    throw IllegalArgumentException(
+                        "The base URI should only contain the scheme and host."
+                    )
```

Comment:
`check` already throws an `IllegalStateException`

```suggestion
                    "The base URI should only contain the scheme and host."
```

Reply:
Good point! Let's address this in a follow-up, I'd like to avoid restarting CI.